src/tests/test_collection.c

```diff
@@ -26,24 +27,28 @@ MU_TEST(test_collection_add_relation)
   mu_assert_int_eq(0, dlite_instance_get_dimension_size((DLiteInstance *)coll,
 							"nrelations"));
 
-  mu_check(!dlite_collection_add_relation(coll, "dog", "is_a", "animal"));
-  mu_check(!dlite_collection_add_relation(coll, "cat", "is_a", "animal"));
-  mu_check(!dlite_collection_add_relation(coll, "terrier", "is_a", "dog"));
+  mu_check(!dlite_collection_add_relation(coll, "dog", "is_a", "animal",
+                                          NULL));
+  mu_check(!dlite_collection_add_relation(coll, "cat", "is_a", "animal",
+                                          NULL));
+  mu_check(!dlite_collection_add_relation(coll, "terrier", "is_a", "dog",
+                                          NULL));
   mu_assert_int_eq(3, dlite_instance_get_dimension_size((DLiteInstance *)coll,
 							"nrelations"));
   mu_assert_int_eq(3, coll->nrelations);
 
   /* dublicate... */
-  mu_check(!dlite_collection_add_relation(coll, "terrier", "is_a", "dog"));
+  mu_check(!dlite_collection_add_relation(coll, "terrier", "is_a", "dog",
+                                          NULL));
```

Comment:
Plase also test setting the datatype.

Reply:
Datatype is already tested in the test_collection_find() function.